### DISCUSSION TITLE
Using Bootstrap v5.2.0 with Understrap

### DISCUSSION THREAD

**Discussion by hsankala** [holger1411/understrap#1853] [Q&A] [answered]
I was looking to use the latest compatible version of bootstrap5 with understrap. 

The most recent stable Bootstrap 5 version is currently [v5.2.0](https://www.npmjs.com/package/bootstrap).

I don’t understand why the child theme package.json isn’t pulling in the latest bootstrap build of v5.2.  The [child theme package.json is set to "bootstrap": "^5.1.3"](https://github.com/understrap/understrap-child/blob/fa849f7a6a9c4ef885b94873fd66b99e2e872ab4/package.json#L55).  Having the [caret](https://stackoverflow.com/a/22345808/2373058) ^5.1.3 should be telling NPM to pulling down latest minor version which is v5.2.0 so the npm install should be pulling down v5.2.0 into the node modules folder.  Am I correct in my understanding here?

I could of course just change package.json in the child theme to pull down bootstrap v5.2, is this version currently compatible with understrap?

**Comment by IanDelMar** [comment, marked as answer]
The Bootstrap version is locked to 5.1.3 via the package-lock.json file (see https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json for a description of the purpose of package-lock.json). Additional to manually changing the package.json you can either run `npm update` (updates all dependencies) or `npm update bootstrap` (updates the Bootstrap dependency only) or delete the package-lock.json  before running `npm install` (updates all dependencies). After installing/updating the packages you have to run `npm copy-assets`.

Understrap and it's child theme are (almost) compatible with Bootstrap 5.2.0. If you use a carousel you have to apply the following changes

```diff
- <div id="carouselExampleControls" class="carousel slide" data-ride="carousel" data-bs-ride="carousel" data-interval="false" data-bs-interval="false">
+ <div id="carouselExampleControls" class="carousel slide" data-interval="false" data-bs-ride="false">
```

**Reply by hsankala** [reply]
Ahh I see, it's locked in the package-lock.json.  Thanks @IanDelMar very clear, thanks additional info like 

```
npm update bootstrap
npm copy-assets
```

And thanks for the heads up about the carousel.

**Comment by peter-menyes** [comment]
The answer is great, however for me (updating in the child-theme) there was one step missing: importing _maps.scss which is new in BS 5.2.0 - otherwise compiling breaks:

> Error: Undefined variable.
> 20 │   @each $color, $value in $theme-colors-rgb {
> src\sass\assets\bootstrap5\_root.scss 20:27  @import

So the solution for me was adding this line to custom-editor-style.scss (after functions and variables):

> @import "assets/bootstrap5/maps";

See bug and fix here: https://github.com/twbs/bootstrap/issues/36785#issuecomment-1189969247

**Reply by IanDelMar** [reply]
You're absolutely right. Forgot to mention that. There should not be any additional necessary changes. Here's a PR which has already been merged into the develop branch updating the Bootstrap version (of the parent theme): https://github.com/understrap/understrap/pull/1821. I had no problems using it with those changes (sidebar-hero.php and custom-editor-style.scss). Thank you for adding the missing info.
